spec/awful/completion_spec.lua

```diff
@@ -5,18 +5,29 @@ local shell = function(...)
 end
 local gfs = require("gears.filesystem")
 
+local test_dir
+
+--- Get and create a temporary test dir based on `pat`, where %d gets replaced by
+-- the current PID.
+local function get_test_dir(pat)
+    local pid = io.open("/proc/self/stat", "r"):read():gmatch("%S+")()
```

Comment:
Linuxism, not good.

I suggest `https://developer.gnome.org/gio/stable/GFile.html#g-file-new-tmp` and `https://developer.gnome.org/gio/stable/GFile.html#g-file-get-path`. However, that only gets you a file and not a directory... remove that file and create a directory in its place?